src/core/or/circuitpadding_machines.c

```diff
@@ -103,88 +92,36 @@ setup_burst_state_for_hiding_intro_circuits(circpad_state_t *burst_state,
    * rely on the state length sampling and not the tokens. */
   burst_state->token_removal = CIRCPAD_TOKEN_REMOVAL_NONE;
 
-  /* Histogram for BURST state:
-   *
-   * We want a machine that will send up to 20 fake cells over a reasonable
-   * time frame, to obfuscate the fact that introduction circuits always send 4
-   * cells both ways */
-
-  /* Figure out the length of the BURST state so that it's randomized. See
-     definition of these constants for rationale. */
+  /* Figure out the length of the BURST state so that it's randomized. We will
+   * set the histogram such that we don't send any padding from the
+   * origin-side, so just tune these parameteres for the relay-side. */
   burst_state->length_dist.type = CIRCPAD_DIST_UNIFORM;
```

Comment:
I do not see the code that prevents us from using the histogram on the origin-side. It looks like the origin side uses the same state machine event setup, and thus will send the same padding as relay side?

Reply:
It's because I do this for the origin-side:
```
    burst_state->histogram_edges[0] = CIRCPAD_DELAY_INFINITE-1;
    burst_state->histogram_edges[1] = CIRCPAD_DELAY_INFINITE;
```
which is a hacky way to never send padding.

It's hacky because it's not actually an infinity bin, more like a infinity-1 bin. That was necessary since we recently enforced strictly increasing bin orders and I could not have two infinity edges in a row.